### PR TITLE
APS-612: Show applicant guidance when blocked by LAO restriction

### PR DESCRIPTION
**Pull request by vubalasu**
# Context

# Changes in this PR
Show applicant guidance when blocked by LAO restriction
### Before
image

### After
Screenshot 2024-04-12 at 17 30 37

### CODE REVIEW

**Review by pezholio — COMMENTED** [review]
I'm thinking actually the way we handle this error doesn't translate into reality. At the moment, if you try a CRN that is restricted in dev (e.g X698334), the API returns a 403, which bubbles up to the frontend and logs the user out.

**Comment by pezholio** [comment]
Ah, this is because I have the Restricted CRNs permission, which isn't in use for applicants. When I remove that permission, it works as expected 👍

File: server/controllers/apply/applicationsController.test.ts [hunk 1]
```diff
@@ -424,6 +424,7 @@ describe('applicationsController', () => {
           errors: errorsAndUserInput.errors,
           errorSummary: errorsAndUserInput.errorSummary,
           ...errorsAndUserInput.userInput,
+          restrictedPerson: person.crn,
```
Comment (richpjames) [inline server/controllers/apply/applicationsController.test.ts:427]: This value is set to be 'true' in People Controller but here we are setting it to be a CRN

**Review by pezholio — APPROVED** [review]
I think this is OK for now. I've had a look through the design system to see if there's a better way of giving more verbose error guidance, but there doesn't look to be anything better out of the box. Hopefully we can revisit this when we have a designer 👍

File: server/controllers/apply/applicationsController.ts [hunk 2]
```diff
@@ -145,12 +145,13 @@ export default class ApplicationsController {
           })
         }
       }
-
+      const restrictedPerson = req.flash('restrictedPerson')[0]
```
Comment (richpjames) [inline server/controllers/apply/applicationsController.ts:148]: could this error if req.flash doesn't have an entry for 'restrictedPerson'?

**Review by richpjames — CHANGES_REQUESTED** [review]

File: server/controllers/apply/applicationsController.test.ts (hunk 1 above)
Comment (vubalasu) [reply server/controllers/apply/applicationsController.test.ts]: updated

File: server/controllers/apply/applicationsController.ts (hunk 2 above)
Comment (vubalasu) [reply server/controllers/apply/applicationsController.ts:148]: gets undefined

**Review by richpjames — APPROVED** [review]

**Pull request merged**